pkg/apis/metal3/v1alpha1/baremetalhost_types.go

```diff
@@ -634,6 +638,11 @@ func (host *BareMetalHost) getLabel(name string) string {
 	return host.Labels[name]
 }
 
+// HasBMCDetails returns true if the BMC details are set
+func (host *BareMetalHost) HasBMCDetails() bool {
+	return host.Spec.BMC.Address != "" || host.Spec.BMC.CredentialsName != ""
```

Comment:
looks like this assumes that host.spec.BMC exists (not `nil`). is this a valid assumption?

Reply:
That field isn't a pointer, so it should be OK.